lib/units/websocket/index.js

```diff
@@ -305,6 +305,28 @@ module.exports = function(options) {
 
     new Promise(function(resolve) {
       socket.on('disconnect', resolve)
+        // Global messages for all clients using socket.io
+        //
+        // Device note
+        .on('device.note', function(data) {
+          return dbapi.setDeviceNote(data.serial, data.note)
+            .then(function() {
+              return dbapi.loadDevice(data.serial)
+            })
+            .then(function(device) {
+              if(device) {
+                io.emit('device.change', {
```

Comment:
Yeah, I also had this in mind that it will not update device note for the users connected with other web-socket server instances. And wanted to discuss this issue.

I do wanted to updated database through processor but current processor design won't let do it. In my first version, I actually made a device "note" plugin which was doing nothing just sending a global message that device note update request is received and then processor will update the db and send changes to all the clients. 

But then I realised this will not work if device is offline, since device unit for that device will be dead. And for this feature, people should at least can update note even if device is offline. So, I finally went through socket.io hack.


I also thought about a different work around. To use rethinkdb `table.change()` method. We can set a callback whenever device table is changed and send a global message(`device.change`) to all the clients. But for this we need to add a new publisher in db unit.

But anyways, processor unit needs an update and should also process client side messages.